### ISSUE TITLE
Ensure braids implement StructuralEq and StructuralPartialEq

### ISSUE THREAD

**Issue by neoeinstein** [neoeinstein/aliri_braid#8]
This will allow for using const values in pattern matching when that lands, as in the following:

```
#[braid]
struct MyData;

const TEST_VAL: &MyDataRef = MyDataRef::from_str("test");

match some_data_ref {
    TEST_VAL => println!("test!"),
    _ => println!("not a test…"),
}
```

**Comment by neoeinstein** [comment]
Still working toward this. Even with const values, the compiler doesn't like this custom type as the pattern to assert on:

```
cannot use unsized non-slice type `Type` in constant patterns
```

See rust-lang/rust#87046

**Comment by neoeinstein** [comment]
This is closed via #13, but the compiler still prevents this as mentioned above.
